src/odbc_result.cpp

```diff
@@ -16,6 +16,8 @@ odbc_result::odbc_result(
       bound_(false),
       output_encoder_(Iconv(c_->encoding(), "UTF-8")) {
 
+  c_->cancel_current_result(false);
```

Comment:
Previously we were only cancelling the results _after_ performing the query which doesn't work when `immediate = TRUE`. I think cancelling before makes more sense, and it's a bit nicer to explicit about it, rather than assuming you remember that `set_current_result` will cancel if needed.